WeakAuras/Prototypes.lua

```diff
@@ -6602,21 +6602,162 @@ Private.event_prototypes = {
+          for i = 1, C_LossOfControl.GetActiveLossOfControlDataCount() do
+            local data = C_LossOfControl.GetActiveLossOfControlData(i)
+            if data then
+              if (not use_controlType)
+              or (data.locType == "PACIFYSILENCE" and (controlType == "SILENCE" or controlType == "PACIFY"))
+              or (data.locType == "STUN_MECHANIC" and controlType == "STUN")
+              or (data.locType == controlType and (controlType ~= "SCHOOL_INTERRUPT" or ((not use_interruptSchool) or data.lockoutSchool == interruptSchool)))
+              then
+                spellId = data.spellID
+                spellName, _, icon = GetSpellInfo(data.spellID)
+                duration = data.duration
+                expirationTime = data.startTime + data.duration
+                locType = data.locType
+                lockoutSchool = data.lockoutSchool
+                controlText = data.displayText
+                show = true
+                break
+              end
+            end
+          end
+      ]=]
+      ret = ret:format(
+        trigger.use_controlType and "true" or "false",
+        type(trigger.controlType) == "string" and "[["..trigger.controlType.."]]" or [["STUN"]],
+        trigger.use_inverse and "true" or "false",
+        trigger.use_interruptSchool and "true" or "false",
+        trigger.interruptSchool or 0
+      )
+      return ret
+    end,
     args = {
       {
-        name = "controlled",
-        display = L["Crowd Controlled"],
-        type = "tristate",
-        init = "not HasFullControl()"
-      }
+        name = "controlType",
+        display = L["Specific CC Type"],
+        type = "select",
+        values = "loss_of_control_Types",
+        conditionType = "select",
+        test = "true",
+        default = "STUN",
+      },
+      {
+        name = "interruptSchool",
+        display = L["Interrupt School"],
+        type = "select",
+        values = "main_spell_schools",
+        conditionType = "select",
+        default = 1,
+        conditionType = "select",
+        test = "true",
+        enable = function(trigger) return trigger.controlType == "SCHOOL_INTERRUPT" end,
+      },
+      {
+        name = "inverse",
+        display = L["Inverse"],
+        type = "toggle",
+        test = "true",
+      },
+      {
+        name = "name",
+        display = L["CC Name"],
+        hidden = true,
+        conditionType = "string",
+        init = "controlText",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "spellName",
+        display = L["CC Spell Name"],
+        hidden = true,
+        conditionType = "string",
+        init = "spellName",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "locType",
+        display = L["CC Type"],
+        hidden = true,
+        conditionType = "string",
+        init = "locType",
```

Comment:
You don't have to use init here.

Reply:
How does the value get stored without it?

Reply:
It should work with just store = true if you have a local variable of that name.

Reply:
Ho! nice.